File: HaskellListas/problema11.hs
{-
Crie uma função com assinatura primeiros :: Int -> [t] -> [t], a qual recebe um número de ele-
mentos, uma lista, e retorna uma lista. Esta função deve retornar uma lista com os n primeiros elementos
informados no primeiro parâmetro. Não utilize nenhuma função pronta to Haskell para esta tarefa.
-}
primeiros :: Int -> [t] -> [t]
primeiros 0 l = []
primeiros i (a:b) = [a] ++ (primeiros (i-1) b)

main = do
  print(primeiros 3 [1,2,3,4,5,6])

File: prolog/listas/problema11.pl
primeiros(0,_,[]).
primeiros(N,[H|T],[H|L2]) :- N>0, N2 is N-1, primeiros(N2,T,L2).

/*
?- primeiros(3,[1,2,3,4,5,6,7],R).
R = [1, 2, 3] .
*/
